src/content/1.7/development/architecture/migration-guide/forms/SQRS-usage-in-forms.md

```diff
@@ -0,0 +1,171 @@
+---
+title: SQRS usage in forms
+weight: 30
+---
+
+# SQRS usage in forms
+
+## Introduction
+
+Assuming that you are already familiar with [SQRS](#) and [CRUD forms]({{< relref "CRUD-form.md" >}}) as this topic only demonstrates the usage of the SQRS approach. To apply it in your forms you need to:
+
+1. Inject a `CommandBus` or `QueryBus` instance using your class constructor.
+2. Call your command using the `CommandBus` or `QueryBus`.
+
+## Example using command bus
+
+To start with, lets inject a `CommandBus` to our _Form data handler_
+
+```yml
+#src/PrestaShopBundle/Resources/config/services/core/form/form_data_handler.yml
+
+prestashop.core.form.identifiable_object.data_handler.contact_form_data_handler:
+    class: 'PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler\ContactFormDataHandler'
+    arguments:
+      - '@prestashop.core.command_bus'
+```
+
+and in `ContactFormDataHandler`
+
+```php
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
+
+final class ContactFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     */
+    public function __construct(CommandBusInterface $commandBus)
+    {
+        $this->commandBus = $commandBus;
+    }
+}
+```
+
+Right now the first step is completed - command bus is injected in the form data handler. Lets use it!
+
+```php
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
+
+final class ContactFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     */
+    public function __construct(CommandBusInterface $commandBus)
+    {
+        $this->commandBus = $commandBus;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws ContactException
+     */
+    public function update($id, array $data)
+    {
+        $editContactCommand = (new EditContactCommand((int) $id))
+            ->setLocalisedTitles($data['title'])
+            ->setEmail($data['email'])
+            ->setIsMessagesSavingEnabled($data['is_messages_saving_enabled'])
+            ->setLocalisedDescription($data['description'])
+            ->setShopAssociation(is_array($data['shop_association']) ? $data['shop_association'] : [])
+        ;
+
+        /** @var ContactId $result */
+        $result = $this->commandBus->handle($editContactCommand);
+
+        return $result->getValue();
+    }
+}
+```
+
+In `update` function command `EditContactCommand` is used to set all required data for further processing. After that, command bus handles given command and in the success case, it returns `ContactId` value object which is used to return contact id.
```

Comment:
this could have simpler explanation i think, something like: "Instead of creating new object directly in `update()` method, you can delegate it to Command. All that you have to do is create command using form `$data` and dispatch it." and it could go above code example imo.